app/controllers/bids_controller.rb

```diff
@@ -130,58 +132,72 @@ def upload_rfq
 
 
     number_of_skipped_rows = 0
+    bid_quotation_for_transporter = BidQuotation.where(bid_id: @bid.id, transporter_id: transporter_id).first
+    if bid_quotation_for_transporter.nil?
+      bid_quotation = BidQuotation.create ({
+             bid_id: @bid.id,
+             bid_quotation_date: Date.today,
+             transporter_id: transporter_id
+               })
+    end
+
 
     (13..spreadsheet.last_row).each do |i|
       row =  spreadsheet.row(i)
       warehouse_selection = WarehouseSelection.find(row[0])
-      bid_quotation_in_db = BidQuotation.where(bid_id: @bid.id, transporter_id: transporter_id).first
-      if !bid_quotation_in_db.nil?
-          bid_quotation_in_db = BidQuotationDetail.where(bid_quotation_id: bid_quotation_in_db.id, location_id: warehouse_selection.location_id, warehouse_id: warehouse_selection.warehouse_id).first
+      if !bid_quotation_for_transporter.nil?
+          bid_quotation_in_db = BidQuotationDetail.where(bid_quotation_id: bid_quotation_for_transporter.id, location_id: warehouse_selection.location_id, warehouse_id: warehouse_selection.warehouse_id).first
       end
-
-
+         
+      
       if bid_quotation_in_db
         if row[5].is_a?( Numeric) &&  row[5] >= 0 && !warehouse_selection.nil?
           bid_quotation_in_db.tariff=row[5]
           bid_quotation_in_db.save
-        else
-          Rails.logger.info("No Warehouse allocation was found for the worda id: #{warehouse_selection.location_id}. Skipping...")
+        
+
         end
 
       else
 
-         if row[5].is_a?( Numeric) &&  row[5] >= 0 && !warehouse_selection.nil? 
-             bid_quotation = BidQuotation.create ({
-             bid_id: @bid.id,
-             bid_quotation_date: Date.today,
-             transporter_id: transporter_id
-               })
+         if row[5].is_a?( Numeric) &&  row[5] >= 0 && !warehouse_selection.nil?  
```

Comment:
@nathnael again what does row[5] stand for? what is it? why do we have to use a fixed field index?